### PR TITLE
Update robot interface

### PR DESCRIPTION
**Pull request by vetlek**
Update robot interface to reflect changes in 'equinor/isar'

### CODE REVIEW

File: src/isar_turtlebot/turtlebot/turtlebot.py [hunk 1]
```diff
@@ -50,67 +37,28 @@ def __init__(self, bridge: RosBridge) -> None:
         self.task_handler: Optional[TaskHandler] = None
 
         self.filenames: dict = dict()
+        self.metadata: dict = dict()
```
Comment (sondreo) [inline src/isar_turtlebot/turtlebot/turtlebot.py:40]:
```suggestion
        self.inspections: dict = dict()
```
Could this be renamed to `inspections`, as it stores the inspection instances returned from `task_handler.get_inspection()`?

File: src/isar_turtlebot/turtlebot/turtlebot.py [hunk 2]
```diff
@@ -50,67 +37,28 @@ def __init__(self, bridge: RosBridge) -> None:
...
-    def get_robot_pose(self) -> Pose:
-        pose_message: dict = self.bridge.pose.get_value()
-        pose: Pose = decode_pose_message(pose_message=pose_message)
-        return pose
-
-    def register_inspection_id(self, inspection_id: UUID) -> None:
-        self.filenames[inspection_id] = self.task_handler.filename
+    def get_inspections(self, id: UUID) -> Sequence[Inspection]:
+        inspection: Inspection = self.metadata[id]
+        inspection.data = self._read_data(id)
```
Comment (sondreo) [inline src/isar_turtlebot/turtlebot/turtlebot.py:57]: Consider adding error handling when retrieving inspection data, like `FileNotFoundError` during reading or `KeyError` if the inspection ID does not match any filename indices.

**Review by sondreo — CHANGES_REQUESTED** [review]
Great work! See some minor suggestions.

File: src/isar_turtlebot/turtlebot/turtlebot.py (hunk 2 above)
Comment (vetlek) [reply src/isar_turtlebot/turtlebot/turtlebot.py:57]: Good suggestion!

File: src/isar_turtlebot/turtlebot/taskhandlers/takethermalimage.py [hunk 3]
```diff
@@ -82,7 +118,7 @@ def _move_status(self) -> Status:
         return move_status
 
     def _write_image_bytes(self):
-        image_data: str = self._get_image_data()
+        image_data: str = self.bridge.visual_inspection.get_image()
```
Comment (sondreo) [inline src/isar_turtlebot/turtlebot/taskhandlers/takethermalimage.py:121]: The thermal image conversion should be added after it was removed from `robot_interface.py`. Consider adding it here, or await #43.

Comment (sondreo) [reply src/isar_turtlebot/turtlebot/taskhandlers/takethermalimage.py:121]: Will be resolved by handling merge conflicts before merge.

**Review by sondreo — APPROVED** [review]
Looks good :+1:

Make sure to include the thermal image conversion from #43 during merge: https://github.com/equinor/isar-turtlebot/blob/6c298b324b20d6fe225e149d6eaf65e96151d358/src/isar_turtlebot/turtlebot/taskhandlers/takethermalimage.py#L108-L114

**Pull request merged**